packages/platform-android/src/commands/runAndroid/tryLaunchEmulator.ts

```diff
@@ -0,0 +1,61 @@
+import {execFileSync, spawn} from 'child_process';
+import Adb from './adb';
+
+const getEmulators = () => {
+  try {
+    const emulatorsOutput = execFileSync('emulator', ['-list-avds']).toString();
+    return emulatorsOutput.split('\n').filter(name => name !== '');
+  } catch {
+    return [];
+  }
+};
+
+const launchEmulator = async (emulatorName: string, adbPath: string) => {
+  return new Promise((resolve, reject) => {
+    const cp = spawn('emulator', [`@${emulatorName}`], {
+      detached: true,
+      stdio: ['ignore', 'pipe', 'ignore'],
+    });
+
+    // Reject command after timeout
+    const rejectTimeout = setTimeout(() => {
+      cp.stdout.destroy();
```

Comment:
Do we need this?

Reply:
Yes, without this I was unable to detach the child process from its parent. The main process `react-native run-android` has not beeing finished until emulator was turned off. Destroying the stdout is disconnecting a pipe between processes and they can be managed independently